generator/booking_dataset.cpp

```diff
@@ -306,58 +308,34 @@ void BookingDataset::LoadHotels(istream & src, string const & addressReferencePa
   }
 }
 
-bool BookingDataset::MatchWithBooking(OsmElement const & e) const
+size_t BookingDataset::MatchWithBooking(FeatureBuilder1 const & fb) const
 {
-  string name;
-  for (auto const & tag : e.Tags())
-  {
-    if (tag.key == "name")
-    {
-      name = tag.value;
-      break;
-    }
-  }
+  auto const name = fb.GetName(StringUtf8Multilang::kDefaultCode);
 
   if (name.empty())
     return false;
 
+  auto const center = MercatorBounds::ToLatLon(fb.GetKeyPoint());
   // Find |kMaxSelectedElements| nearest values to a point.
   auto const bookingIndexes =
-      GetNearestHotels(e.lat, e.lon, kMaxSelectedElements, kDistanceLimitInMeters);
-
-  bool matched = false;
+      GetNearestHotels(center.lat, center.lon, kMaxSelectedElements, kDistanceLimitInMeters);
```

Comment:
Why not to pass center to GetNearestHotels? What's the point to pass individually lat and lon?